<robot name="single_lwr_robot">
  
  
  
  
  <material name="Kuka/DarkGrey">
    <color rgba="0.3 0.3 0.3 1.0" />
  </material>
  <material name="Kuka/LightGrey">
    <color rgba="0.8 0.8 0.8 1.0" />
  </material>
  <material name="Kuka/Red">
    <color rgba="0.3 0.3 0.3 1.0" />
  </material>
  <material name="Kuka/Black">
    <color rgba="0 0 0 1.0" />
  </material>
  <material name="Kuka/Orange">
    <color rgba="1.0 0.487 0 1.0" />
  </material>
  <material name="Kuka/Silver">
    <color rgba="1.0 1.0 1.0 1.0" />
  </material>
  <material name="Kuka/Grey">
    <color rgba="0.7 0.7 0.7 1.0" />
  </material>
  <material name="Kuka/Blue">
    <color rgba="0.0 0.0 0.8 1.0" />
  </material>
  
  
  
  
  
  <link name="box">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.50" />
      <geometry>
        <box size="1.20 .74 1.0" />
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.50" />
      <geometry>
        <box size="1.20 .74 1.0" />
      </geometry>
    </collision>
  </link>
  
  
  <joint name="box_lwr_base_joint" type="fixed">
    <origin rpy="0 0 0" xyz="-0.25 0 1" />
    <origin rpy="0 0 0" xyz="0 0 0" />
    <parent link="box" />
    <child link="lwr_base_link" />
  </joint>
  <link name="lwr_base_link">
    <inertial>
      <mass value="1.58331084746" />
      <origin xyz="-0.0213141013778 -0.0341782643576 -0.0104355130861" />
      <inertia ixx="-0.0831421287841" ixy="-0.0281205823292" ixz="-0.0392307373339" iyy="0.0832844086014" iyz="0.0164316150208" izz="-0.00722685865565" />
    </inertial>
    <visual>
      <origin rpy="0 0 3.14159265359" xyz="0.001 0.001 0.001" />
      <geometry>
        <mesh filename="package://lwr_description/meshes/base.STL" />
      </geometry>
      <material name="Kuka/Orange" />
    </visual>
    <collision>
      <origin rpy="0 0 3.14159265359" xyz="0.001 0.001 0.055" />
      <geometry>
        
        <cylinder length="0.11" radius="0.07" />
      </geometry>
    </collision>
  </link>
  
  <joint name="lwr_0_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.11" />
    <axis xyz="0 0 1" />
    <limit effort="176" lower="-2.96705972839" upper="2.96705972839" velocity="1.91986217719" />
    <dynamics damping="1.0" friction="10.0" />
    <parent link="lwr_base_link" />
    <child link="lwr_1_link" />
  </joint>
  <link name="lwr_1_link">
    <inertial>
      <mass value="2.832425315" />
      <origin rpy="0 0 0" xyz="0.0041184903736 -0.0424025287785 0.139802204154" />
      <inertia ixx="0.0633371509729" ixy="0.00163267958473" ixz="0.00637913924478" iyy="-0.0830495134728" iyz="0.00200212918117" izz="0.108909059849" />
    </inertial>
    <visual>
      <origin rpy="0 0 3.14159265359" xyz="0.001 0.001 -0.008" />
      <geometry>
        <mesh filename="package://lwr_description/meshes/link_1.STL" />
      </geometry>
      <material name="Kuka/Orange" />
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.001 0.001 0.0" />
      <geometry>
        
        <sphere radius="0.07" />
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 0 0" xyz="0.001 0.06 0.2" />
      <geometry>
        <sphere radius="0.07" />
      </geometry>
    </collision>
    <collision>
      <origin rpy="-0.291 0 0" xyz="0.001 0.03 0.1" />
      <geometry>
        <cylinder length="0.208" radius="0.07" />
      </geometry>
    </collision>
  </link>
  
  <joint name="lwr_1_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0.001 0.001 0.2005" />
    <axis xyz="0 -1 0" />
    <limit effort="176" lower="-2.09439510239" upper="2.09439510239" velocity="1.91986217719" />
    <dynamics damping="1.0" friction="10.0" />
    <parent link="lwr_1_link" />
    <child link="lwr_2_link" />
  </joint>
  <link name="lwr_2_link">
    <inertial>
      <mass value="2.6981578399" />
      <origin rpy="0 0 0" xyz="0.0244536299689 0.0857602351445 0.0547207728664" />
      <inertia ixx="0.036290795558" ixy="0.0410533772091" ixz="0.0160608265987" iyy="-0.0148992038497" iyz="0.0558089817272" izz="-0.0569035915511" />
    </inertial>
    <visual>
      <origin rpy="0 0 3.14159265359" xyz="0 0 0.0" />
      <geometry>
        <mesh filename="package://lwr_description/meshes/link_2.STL" />
      </geometry>
      <material name="Kuka/Orange" />
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.2" />
      <geometry>
        
        <sphere radius="0.07" />
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 0 0" xyz="0 -0.06 0.0" />
      <geometry>
        <sphere radius="0.07" />
      </geometry>
    </collision>
    <collision>
      <origin rpy="-0.291 0 0" xyz="0 -0.03 0.1" />
      <geometry>
        <cylinder length="0.208" radius="0.07" />
      </geometry>
    </collision>
  </link>
  
  <joint name="lwr_2_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.20" />
    <axis xyz="0 0 1" />
    <limit effort="100" lower="-2.96705972839" upper="2.96705972839" velocity="2.23402144255" />
    <dynamics damping="1.0" friction="10.0" />
    <parent link="lwr_2_link" />
    <child link="lwr_3_link" />
  </joint>
  <link name="lwr_3_link">
    <inertial>
      <mass value="2.64023810144" />
      <origin rpy="0 0 0" xyz="0.0171124024251 0.0535237057985 0.165883167002" />
      <inertia ixx="-0.0785075497892" ixy="0.0136192343462" ixz="0.0111956803489" iyy="0.00531802796702" iyz="0.0165267588882" izz="-0.00088074512262" />
    </inertial>
    <visual>
      <origin rpy="0 0 3.14159265359" xyz="0 0 -0.008" />
      <geometry>
        <mesh filename="package://lwr_description/meshes/link_3.STL" />
      </geometry>
      <material name="Kuka/Orange" />
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.0" />
      <geometry>
        
        <sphere radius="0.07" />
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 0 0" xyz="0 -0.06 0.2" />
      <geometry>
        <sphere radius="0.07" />
      </geometry>
    </collision>
    <collision>
      <origin rpy="0.291 0 0" xyz="0 -0.03 0.1" />
      <geometry>
        <cylinder length="0.208" radius="0.07" />
      </geometry>
    </collision>
  </link>
  
  <joint name="lwr_3_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.20" />
    <axis xyz="0 1 0" />
    <limit effort="100" lower="-2.09439510239" upper="2.09439510239" velocity="2.23402144255" />
    <dynamics damping="1.0" friction="10.0" />
    <parent link="lwr_3_link" />
    <child link="lwr_4_link" />
  </joint>
  <link name="lwr_4_link">
    <inertial>
      <mass value="2.7132577627" />
      <origin rpy="0 0 0" xyz="0.00531164774391 -0.0985406336186 0.0757159948899" />
      <inertia ixx="-0.0569589959373" ixy="-0.0806008047504" ixz="-0.0227033808722" iyy="-0.000514513758388" iyz="-0.0297940102933" izz="0.0770074450322" />
    </inertial>
    <visual>
      <origin rpy="0 0 3.14159265359" xyz="0 0 0.0" />
      <geometry>
        <mesh filename="package://lwr_description/meshes/link_4.STL" />
      </geometry>
      <material name="Kuka/Orange" />
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.2" />
      <geometry>
        
        <sphere radius="0.07" />
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 0 0" xyz="0 0.06 0.0" />
      <geometry>
        <sphere radius="0.07" />
      </geometry>
    </collision>
    <collision>
      <origin rpy="0.291 0 0" xyz="0 0.03 0.1" />
      <geometry>
        <cylinder length="0.208" radius="0.07" />
      </geometry>
    </collision>
  </link>
  
  <joint name="lwr_4_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.20" />
    <axis xyz="0 0 1" />
    <limit effort="100" lower="-2.96705972839" upper="2.96705972839" velocity="3.56047167407" />
    <dynamics damping="1.0" friction="10.0" />
    <parent link="lwr_4_link" />
    <child link="lwr_5_link" />
  </joint>
  <link name="lwr_5_link">
    <inertial>
      <mass value="1.73773034112" />
      <origin rpy="0 0 0" xyz="-0.0236322488631 -0.0171223500763 0.105710870931" />
      <inertia ixx="0.018823299231" ixy="-0.00427103757604" ixz="-0.0553112177034" iyy="0.0641555364238" iyz="0.0245031037595" izz="0.089122970362" />
    </inertial>
    <visual>
      <origin rpy="0 0 3.14159265359" xyz="0 0 -0.008" />
      <geometry name="lwr_5_geom">
        <mesh filename="package://lwr_description/meshes/link_5.STL" />
      </geometry>
      <material name="Kuka/Orange" />
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.0" />
      <geometry>
        
        <sphere radius="0.07" />
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 0 0" xyz="0 0.04 0.19" />
      <geometry>
        <sphere radius="0.07" />
      </geometry>
    </collision>
    <collision>
      <origin rpy="-0.207 0 0" xyz="0 0.02 0.095" />
      <geometry>
        <cylinder length="0.198" radius="0.07" />
      </geometry>
    </collision>
  </link>
  
  <joint name="lwr_5_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.19" />
    <axis xyz="0 -1 0" />
    <limit effort="38" lower="-2.09439510239" upper="2.09439510239" velocity="3.21140582367" />
    <dynamics damping="1.0" friction="10.0" />
    <parent link="lwr_5_link" />
    <child link="lwr_6_link" />
  </joint>
  <link name="lwr_6_link">
    <inertial>
      <mass value="1.56394365381" />
      <origin rpy="0 0 0" xyz="0.00974632703136 0.0685903903126 0.106375115445" />
      <inertia ixx="0.0308200720882" ixy="0.0399667013804" ixz="0.037163911402" iyy="0.0557283425191" iyz="0.114819049054" izz="0.0263958385884" />
    </inertial>
    <visual>
      <origin rpy="0 0 3.14159265359" xyz="0 0 0" />
      <geometry>
        <mesh filename="package://lwr_description/meshes/link_6.STL" />
      </geometry>
      <material name="Kuka/Grey" />
    </visual>
    <collision>
      <origin rpy="0 0 3.14159265359" xyz="0 0 0" />
      <geometry>
        <mesh filename="package://lwr_description/meshes/convex/arm_wrist_convex.stl" />
      </geometry>
    </collision>
  </link>
  
  <joint name="lwr_6_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.078" />
    <axis xyz="0 0 1" />
    <limit effort="38" lower="-2.96705972839" upper="2.96705972839" velocity="3.21140582367" />
    <dynamics damping="1.0" friction="10.0" />
    <parent link="lwr_6_link" />
    <child link="lwr_7_link" />
  </joint>
  <link name="lwr_7_link">
    <inertial>
      <mass value="0.382724970432" />
      <origin xyz="0.185107273414 -0.129851143561 0.116388492473" />
      <inertia ixx="0.0243496593734" ixy="0.0141824202012" ixz="0.0212046146686" iyy="0.0747912197253" iyz="0.00626573350136" izz="-0.0847957647384" />
    </inertial>
    <visual>
      <origin rpy="0 0 3.14159265359" xyz="0 0 0" />
      <geometry>
        <mesh filename="package://lwr_description/meshes/link_7.STL" />
      </geometry>
      <material name="Kuka/DarkGrey" />
    </visual>
    <collision>
      <origin rpy="0 0 3.14159265359" xyz="0 0 0" />
      <geometry>
        <cylinder length="0.05" radius="0.04" />
      </geometry>
    </collision>
  </link>
  <transmission name="lwr_0_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="lwr_0_joint">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
      <hardwareInterface>EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="lwr_0_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="lwr_1_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="lwr_1_joint">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
      <hardwareInterface>EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="lwr_1_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="lwr_2_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="lwr_2_joint">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
      <hardwareInterface>EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="lwr_2_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="lwr_3_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="lwr_3_joint">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
      <hardwareInterface>EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="lwr_3_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="lwr_4_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="lwr_4_joint">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
      <hardwareInterface>EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="lwr_4_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="lwr_5_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="lwr_5_joint">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
      <hardwareInterface>EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="lwr_5_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="lwr_6_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="lwr_6_joint">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
      <hardwareInterface>EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="lwr_6_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
</robot>